Packs/Code42/Integrations/Code42EventCollector/Code42EventCollector.py

```diff
@@ -0,0 +1,400 @@
+        """
+        demisto.debug(f'Audit logs: {start_time=}, {end_time=}, {limit=}')
+        audit_logs = []
+
+        for audit_log in self.code42_client.audit_log.v1.iter_all(
+            start_time=start_time, end_time=end_time, page_size=page_size  # iterates all the pages
+        ):
+            encoded_audit_log = json.dumps(audit_log, sort_keys=True).encode()
+            audit_log["id"] = hashlib.sha256(encoded_audit_log).hexdigest()
+            audit_log["_time"] = dateparser.parse(audit_log["timestamp"])
+            audit_log["eventType"] = EventType.AUDIT
+            audit_logs.append(audit_log)
+
+        audit_logs = sorted(
+            audit_logs, key=lambda _log: _log["_time"]  # type: ignore[arg-type, return-value]
+        )
+        return audit_logs[:limit]
+
+    def get_file_events(
+        self,
+        start_time: datetime | str | timedelta,
+        end_time: datetime | str | timedelta | None = None,
+        limit: int = MAX_FETCH_FILE_EVENTS,
+        page_size: int = MAX_FILE_EVENTS_PAGE_SIZE
+    ) -> List[dict[str, Any]]:
+        """
+        Get file events
+
+        Args:
+            start_time: from which start time to get events
+            end_time: until which time to get events
+            limit: maximum events to retrieve
+            page_size: the page size per single request
+        """
+        demisto.debug(f'File Events: {start_time=}, {end_time=}, {limit=}')
+        query = EventQuery(start_date=start_time, end_date=end_time, srtDir="asc", pgSize=page_size)
+        response = self.code42_client.file_events.v2.search(
+            query
+        )
+
+        if response.total_count == 0:
+            return []
+        file_events = response.file_events
+        while query.page_token is not None and len(file_events) < limit:
+            response = self.code42_client.file_events.v2.search(query)
+            if current_events := response.file_events:
+                file_events.extend(current_events)
+
+        file_events = file_events[:limit]
+
+        for event in file_events:
+            event.eventType = EventType.FILE
+            event._time = event.event.inserted
+
+        return [event.dict() for event in file_events]
+
+
+def dedup_fetched_events(
+    events: List[dict],
+    last_run_fetched_event_ids: Set[str],
+    keys_list_to_id: List[str]
+) -> List[dict]:
+    """
+    Dedup events, removes events which were already fetched.
+
+    Args:
+        events: the events to deduplicate
+        last_run_fetched_event_ids: a list of already fetched IDs from previous run
+        keys_list_to_id (list): a list of keys to retrieve the ID from the event
+    """
+    un_fetched_events = []
+
+    for event in events:
+        event_id = dict_safe_get(event, keys=keys_list_to_id)
+        event_type = event["eventType"]
+        if event_id not in last_run_fetched_event_ids:
+            demisto.debug(f'event {event_type} with ID {event_id} has not been fetched.')
```

Comment:
instead of logging each line lets add to a list and log it all. will be easier to read